cmd/kubectl-gadget/trace/tcpdump.go

```diff
@@ -0,0 +1,244 @@
+	var decoderArgsParam string
+	var decoderBinaryParam string
+	var snapLen int
+	var filenameParam string
+
+	cmd := &cobra.Command{
+		Use:   "tcpdump",
+		Short: "Trace packets",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			decoderArgs := []string{}
+			var decoder Decoder
+			var decoderBinary string
+			var ngw *pcapgo.NgWriter
+
+			// Writer, used for external output (pcapng)
+			var out io.Writer
+
+			switch Decoder(decoderParam) {
+			case DecoderWireshark:
+				decoder = DecoderExternal
+				decoderArgs = []string{"-k", "-i", "-"}
+				decoderBinary = "wireshark"
+			case DecoderTCPDump:
+				decoder = DecoderExternal
+				decoderArgs = []string{"-r", "-"}
+				decoderBinary = "tcpdump"
+			case DecoderInternal:
+				decoder = DecoderInternal
+			case DecoderFile:
+				decoder = DecoderFile
+			default:
+				return errors.New("unknown decoder")
+			}
+
+			if decoder == DecoderFile {
+				if filenameParam == "" {
+					return fmt.Errorf("no filename specified")
+				}
+				f, err := os.Create(filenameParam)
+				if err != nil {
+					return fmt.Errorf("creating file: %w", err)
+				}
+				out = f
+			}
+
+			if decoder == DecoderExternal {
+				r, w, err := os.Pipe()
+				if err != nil {
+					return fmt.Errorf("could not create pipe: %w", err)
+				}
+
+				if decoderArgsParam != "" {
+					decoderArgs = append(decoderArgs, strings.Split(decoderArgsParam, " ")...)
+				}
+				if decoderBinaryParam != "" {
+					decoderBinary = decoderBinaryParam
+				}
+
+				decoderCmd = exec.Command(decoderBinary, decoderArgs...)
+				decoderCmd.Stdout = os.Stdout
+				decoderCmd.Stderr = os.Stderr
+				decoderCmd.Stdin = r
+				err = decoderCmd.Start()
+				if err != nil {
+					return fmt.Errorf("could not start tcpdump: %w", err)
+				}
+
+				out = w
+			}
+
+			if out != nil {
+				var err error
+				dummyInterface := pcapgo.DefaultNgInterface
+				dummyInterface.LinkType = layers.LinkTypeEthernet
+				dummyInterface.SnapLength = uint32(snapLen)
+				ngw, err = pcapgo.NewNgWriterInterface(out, dummyInterface, pcapgo.NgWriterOptions{SectionInfo: pcapgo.NgSectionInfo{
+					Hardware:    runtime.GOARCH,
+					OS:          runtime.GOOS,
+					Application: "InspektorGadget",
+					Comment:     "using gopacket",
+				}})
+				if err != nil {
+					return fmt.Errorf("instantiating NgWriter: %w", err)
+				}
+				ngw.Flush()
+			}
+
+			filter := strings.Join(args, " ")
+
+			tcpdumpGadget := &TraceGadget[types.Event]{
+				name:        "tcpdump",
+				commonFlags: commonFlags,
+				parser:      NewTCPDump(&commonFlags.OutputConfig, filter, snapLen, decoder, ngw),
+				params: map[string]string{
+					types.FilterStringParam: filter,
+					types.SnapLenParam:      strconv.Itoa(snapLen),
+				},
+			}
+
+			return tcpdumpGadget.Run()
+		},
+	}
+
+	utils.AddCommonFlags(cmd, commonFlags)
+	cmd.Flags().StringVar(&decoderParam, "decoder", "internal", "name of the decoder to use (either tcpdump, wireshark, internal or file)")
```

Comment:
Would use `-o tcpdump`, `-o wireshark`, `-o file=/tmp/foo.pcap`? (It probably requires some extra work but maybe it's cleaner?)